### PR TITLE
add fetch api

### PR DESCRIPTION
**Pull request by plantain-00**
Fixes https://github.com/Microsoft/TypeScript/issues/8966.
The definitions are from https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/whatwg-fetch/whatwg-fetch.d.ts

### CODE REVIEW

**Comment by mhegazy** [comment]
👍

**Comment by mhegazy** [comment]
@zhengbli would this be autogenerated if we took a drop of the Edge definition files today?

**Comment by zhengbli** [comment]
@mhegazy Yes. The latest definite in the redstone build has a `GlobalFetch` interface, and `window` implements it.

File: baselines/dom.generated.d.ts [hunk 1]
```diff
@@ -13855,6 +13856,85 @@ interface JsonWebKey {
...
+    new(input: string|Request, init?:RequestInit): Request;
+}
+
+interface Response extends Body {
+    type: ResponseType;
+    status: number;
+    ok: boolean;
+    statusText: string;
+    headers: Headers;
+    clone(): Response;
```
Comment (zhengbli) [inline baselines/dom.generated.d.ts]: `Request` should also have a `clone` method that returns `Request`

**Comment by mhegazy** [comment]
then we should just take a new drop, instead of hand authoring these definitions..

**Comment by zhengbli** [comment]
As mentioned above, the new spec files that we haven't used yet contains the definition of `fetch` APIs, so it might be more consistent if we wait to take the drop. Although we should speed up the process to something like a monthly update. I'll talk to the Edge team to see if there are anything blocking that from happening. 

Closing this issue for now.

**Pull request closed without merging**